luigi/worker.py

```diff
@@ -407,6 +407,8 @@ def check_complete(task, out_queue):
 class worker(Config):
     # NOTE: `section.config-variable` in the config_path argument is deprecated in favor of `worker.config_variable`
 
+    id = Parameter(default=None,
```

Comment:
`Parameter`s should have a default of `''`. If you truly want default `None`, consider `OptionalParameter`

Reply:
Nice catch @dlstadther, but for this case certainly use `''` and just do `if` on it.

Reply:
Changing this to `''` should be good enough. We are already doing `_config.id or` anyway.